elastic/kibana
# [Security Solution][Exceptions] - Moves remaining exceptions builder logic into lists plugin

## Summary

Moves part of the exceptions UI out of the security solution plugin and into the lists plugin. In order to keep PRs (relatively) small, I am moving single components at a time. This should also then help more easily pinpoint the source of any issues that come up along the way.

The next couple PRs will focus on the exception builder. This one in particular is focused on moving over the `ExceptionBuilderComponent` which deals with rendering numerous exception items and their entries.

Quick Summary:
- `x-pack/plugins/security_solution/public/common/components/exceptions/builder/` → ` x-pack/plugins/lists/public/exceptions/components/builder/`
  - Corresponding unit test file moved as well 
  - Updated security solution exception builder to pull `ExceptionBuilderComponent` from lists plugin

##### Entry Item ([PR](https://github.com/elastic/kibana/pull/94515))
<img width="1022" alt="Screen Shot 2021-03-14 at 11 31 04 PM" src="https://user-images.githubusercontent.com/10927944/111112669-c9bbb580-851d-11eb-85e1-5fff829bd4f0.png">

##### Exception Item  ([PR](https://github.com/elastic/kibana/pull/95246))
<img width="1031" alt="Screen Shot 2021-03-14 at 11 32 15 PM" src="https://user-images.githubusercontent.com/10927944/111112689-d50ee100-851d-11eb-96fc-b32ce51917ea.png">

##### Builder (**_THIS PR DEALS WITH_ 👇🏽**)
![image](https://user-images.githubusercontent.com/10927944/112263040-db8f0e00-8c2b-11eb-8764-a03e3b8cb648.png)

##### Builder Buttons (**_THIS PR DEALS WITH_ 👇🏽**)
<img width="1023" alt="Screen Shot 2021-03-14 at 11 32 39 PM" src="https://user-images.githubusercontent.com/10927944/111112739-e821b100-851d-11eb-8ef5-ac87733d6ef3.png">

##### Exception Item Comments
<img width="1026" alt="Screen Shot 2021-03-14 at 11 33 11 PM" src="https://user-images.githubusercontent.com/10927944/111112766-f7086380-851d-11eb-91f3-d433c0290c6a.png">

##### Builder Modal
<img width="1027" alt="Screen Shot 2021-03-14 at 11 33 30 PM" src="https://user-images.githubusercontent.com/10927944/111112787-02f42580-851e-11eb-95d5-dc968deb0670.png">

### Notes
-  There may still be work to do to get the component working for all use cases, however, for the moment being, concentrating on moving existing functionality over

### Testing
Prior to making these changes, added a number of tests to ensure coverage of existing functionality. For manual testing, please checkout PR and play around with the following:
  - Add/edit an exception item
    - if an existing item, make sure that it populates as expected in the modal 
  - Add multiple exception items (using `or` operator)
  - Play around with the component in storybook

### Checklist
